docs/cn/deploy/Running-Alluxio-On-a-HA-Cluster.md

```diff
@@ -0,0 +1,274 @@
+
+- [方法1](#option1-raft-based-embedded-journal): 使用基于RAFT的内部复制状态机来存储文件系统日志和leading master的选举。这种方法是在Alluxio 2.0中引入的，不需要依赖任何外部服务。
+- [方法2](#option2-zookeeper-and-shared-journal-storage):
+利用外部Zookeeper服务做leading master选举和利用共享存储（例如，root UFS）来共享日志。请参阅[日志管理文档]（{{'/en/operation/Journal.html' | relativize_url}}），以获取更多有关选择和配置Alluxio日记系统信息。
+
+## 前提条件
+
+* 要部署Alluxio群集，首先 下载 预编译的Alluxio二进制文件，解压缩tarball文件并将解压的目录复制到所有节点（包括运行master和worker的所有节点）。
+* 激活不需要密码的从master节点到worker节点的SSH登录。 可以将主机的公共SSH密钥添加到`〜/.ssh/authorized_keys`中。 有关更多详细信息，请[参见本教程](http://www.linuxproblem.org/art_9.html)。
+* 开放所有节点之间的TCP通信。 对于基本功能，确保所有节点上RPC端口都是打开的（default:19998）。
+
+## 基本配置
+### 选项1:基于Raft的嵌入式日志
+
+HA集群的最小配置是将嵌入式日志地址提供给集群内的所有节点。在每个Alluxio节点上，依据模板创建 `conf/alluxio-site.properties` 配置文件。
+
+```console
+$ cp conf/alluxio-site.properties.template conf/alluxio-site.properties
+```
+
+将以下属性添加到 `conf/alluxio-site.properties` 文件:
+
+```properties
+alluxio.master.hostname=<MASTER_HOSTNAME> # 仅在master节点上需要
+alluxio.master.mount.table.root.ufs=<STORAGE_URI>
+alluxio.master.embedded.journal.addresses=<EMBEDDED_JOURNAL_ADDRESS>
+```
+
+说明:
+- 第一个属性`alluxio.master.hostname=<MASTER_HOSTNAME>` 每个master节点上必须是其自身外部可访问主机名。master quorum的每个单独组成部分都需要具有自己的地址集。在worker节点上，此参数将被忽略。示例包括 `alluxio.master.hostname = 1.2.3.4`， `alluxio.master.hostname=node1.a.com`。
+- 第二个属性 `alluxio.master.mount.table.root.ufs=<STORAGE_URI>` 设置为挂载到Alluxio根目录的底层存储URI。 一定保证master节点和所有worker节点都可以访问此共享存储。 示例包括`alluxio.master.mount.table.root.ufs=hdfs://1.2.3.4:9000/alluxio/root/`或`alluxio.master.mount.table.root.ufs=s3://bucket/dir/` 。
+- 第三个属性 `alluxio.master.embedded.journal.addresses` 设置参加Alluxio leading master选举的master节点集。默认的嵌入式日志端口是 `19200`。例如: `alluxio.master.embedded.journal.addresses=master_hostname_1:19200，master_hostname_2:19200，master_hostname_3:19200`
+
+嵌入式日记特性依赖于 [Copycat](https://github.com/atomix/copycat) 内置leader选举功能。内置leader选举功能不能与Zookeeper一起使用，因为系统不能接受两面可能选出不同leaders的状况。启用嵌入式日记就启用了Alluxio的内置leader election机制。请参阅[嵌入式日志配置文档]（{{'/en/operation/Journal.html' | relativize_url}}＃embedded-journal-configuration），以了解更多详细信息以及使用内部leader选举配置HA集群的替代方案。
+
+### 选项2:Zookeeper和共享日志存储
+
+设置Zookeeper HA集群的前提条件:
+1. [ZooKeeper](http://zookeeper.apache.org/) 集群。 Alluxio masters使用ZooKeeper进行leader选举，Alluxio客户端和workers使用ZooKeeper来查询当前谁是leading master。
+1. 存放日志的共享存储系统（所有Alluxio masters均可访问）。leading master在此共享存储系统上写入日志，同时standby masters不断重播日志条目以保持与最新日志同步。建议将日志存储系统设置为:
+  - 高度可用。所有master上元数据修改都需要写日志，因此日志存储系统的任何停机时间都将直接影响Alluxio master的可用性。
+  - 设置共享存储系统为文件系统，而不要用对象存储。因为Alluxio master会写入存储系统的日志文件，和使用文件系统相应操作，例如重命名和flush。对象存储不是不支持这些操作就是即使支持执行速度很慢，因此，如果选择将日志存在对象存储中，Alluxio master的运行吞吐量将大幅降低。
+
+必须设置的最小配置参数是:
+
+```
+alluxio.zookeeper.enabled=true
+alluxio.zookeeper.address=<ZOOKEEPER_ADDRESS>
+alluxio.master.journal.type=UFS
+alluxio.master.journal.folder=<JOURNAL_URI>
+```
+
+说明:
+- 属性 `alluxio .zookeeper.enabled = true masters`启用HA模式，并通知workers已启用HA模式。
```

Comment:
remove the extra space between property key and value. The default way to write `conf/alluxio-site.properties`  is no spaces between `key`, `=`, and `value`. It can be thought as one kind of `code`.

Reply:
The same for all following key-value pairs